roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,159 @@
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._roi_response_fluorescence = self._roi_response
+        self._roi_response_neuropil = self._trace_extractor_read('C')
+        self._roi_response_deconvolved = self._trace_extractor_read('S')
+        self._images_correlation = self._summary_image_read()
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'][()]
+        self.image_masks = self._image_mask_sparse_read()[-1]
+
+    def __del__(self):
+        self._dataset_file.close()
+
+    def _file_extractor_read(self):
+        f = h5py.File(self.filepath, 'r')
+        return f
+
+    def _image_mask_sparse_read(self):
+        roi_ids = self._dataset_file['estimates']['A']['indices']
+        masks = self._dataset_file['estimates']['A']['data']
+        ids = self._dataset_file['estimates']['A']['indptr']
+        _image_mask = np.reshape(csc_matrix((masks, roi_ids, ids), shape=(np.prod(self.get_image_size()),self.no_rois)).toarray(),
+            [self.get_image_size()[0],self.get_image_size()[1],-1],order='F')
+        return masks, roi_ids, ids, _image_mask
+
+    def _trace_extractor_read(self, field):
+        if self._dataset_file['estimates'].get(field):
+            return self._dataset_file['estimates'][field] # lazy read dataset)
+        else:
+            return None
+
+    def _summary_image_read(self):
+        if self._dataset_file['estimates'].get('Cn'):
+            return np.array(self._dataset_file['estimates']['Cn']).T
+        else:
+            return None
+
+    def get_accepted_list(self):
+        accepted = self._dataset_file['estimates']['idx_components']
+        if len(accepted.shape)==0:
+            accepted = list(range(self.no_rois))
+        return accepted
+
+    def get_rejected_list(self):
+        rejected = self._dataset_file['estimates']['idx_components_bad']
+        if len(rejected.shape) == 0:
+            rejected = [a for a in range(self.no_rois) if a not in set(self.get_accepted_list())]
+        return rejected
```

Comment:
should return None if idx_components_bad is missing

Reply:
fixed https://github.com/catalystneuro/roiextractors/pull/34/commits/72eb48defa694c7207d8b218e8f7d85b4164d0af